scheduler/test/cook/test/mesos/scheduler.clj

```diff
@@ -763,5 +766,385 @@
       @(sched/receive-offers offers-chan mock-driver [offer-3])
       (is (= @declined-offer-ids-atom [(:id offer-2) (:id offer-3)])))))
 
+(deftest test-below-quota?
+  (testing "not using quota"
+    (is (sched/below-quota? {:count 5, :cpus 15, :mem 9999}
+                            {:count 0, :cpus 0, :mem 0})))
+  (testing "not using quota with extra keys"
+    (is (sched/below-quota? {:count 5, :cpus 15, :mem 9999, :foo 2, :bar 3}
+                            {:count 0, :cpus 0, :mem 0})))
+  (testing "inside quota"
+    (is (sched/below-quota? {:count 5, :cpus 15, :mem 9999}
+                            {:count 4, :cpus 10, :mem 1234})))
+  (testing "at quota limit"
+    (is (sched/below-quota? {:count 5, :cpus 15, :mem 9999}
+                            {:count 5, :cpus 15, :mem 9999})))
+  (testing "exceed quota limit - count"
+    (is (not (sched/below-quota? {:count 5, :cpus 15, :mem 9999}
+                                 {:count 6, :cpus 10, :mem 1234}))))
+  (testing "exceed quota limit - cpus"
+    (is (not (sched/below-quota? {:count 5, :cpus 15, :mem 9999}
+                                 {:count 4, :cpus 20, :mem 1234}))))
+  (testing "exceed quota limit - mem"
+    (is (not (sched/below-quota? {:count 5, :cpus 15, :mem 1234}
+                                 {:count 4, :cpus 10, :mem 4321}))))
+  (testing "exceed quota limit - gpus"
+    (is (not (sched/below-quota? {:count 5, :cpus 15, :mem 9999, :gpus 4}
+                                 {:count 4, :cpus 10, :mem 1234, :gpus 5}))))
+  (testing "at quota limit with extra keys"
+    (is (sched/below-quota? {:count 5, :cpus 15, :mem 9999, :gpus 4}
+                            {:count 5, :cpus 15, :mem 9999}))))
+
+(deftest test-job->usage
+  (testing "cpus mem and gpus usage"
+    (is (= {:cpus 2, :mem 2048, :count 1} ;; TODO verify whether gpus are intentionally omitted?
```

Comment:
Okay I will create a separate issue and PR for it.

Created: https://github.com/twosigma/Cook/issues/262